pkg/sql/colexec/execgen/template.go

```diff
@@ -89,50 +89,74 @@ func replaceTemplateVars(
 // and a caller
 //   b(true)
 // this function will generate
-//   x = 3
+//   if true {
+//     x = 3
+//   } else {
+//     x = 4
+//   }
 //   return x
+//
+// but because true is a constant, it will be reduced to
+//
+// x = 3
+// return x
```

Comment:
I think some comments about how this enables template variable "passthrough" are needed